enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -549,11 +645,31 @@ func (mc *MigrationCache) Reader(ctx context.Context, r *rspb.ResourceName, unco
 
 	mc.sendNonBlockingCopy(ctx, r, true /*=onlyCopyMissing*/)
 
-	return &doubleReader{
-		src:  srcReader,
-		dest: destReader,
-		r:    r,
-	}, nil
+	dr := &doubleReader{
+		src:          srcReader,
+		dest:         destReader,
+		r:            r,
+		decompressor: decompressor,
+		pw:           pw,
+		pr:           pr,
+		mu:           sync.Mutex{},
+		done:         make(chan struct{}, 0),
+	}
+
+	go func() {
+		defer close(dr.done)
```

Comment:
will this goroutine ever exit if shouldDecompressAndVerify is false?

Reply:
I added a check to only launch the goroutine when shouldDecompressAndVerify is true